slamd/discovery/processing/algorithms/plot_generator.py

```diff
@@ -34,19 +34,33 @@ def create_target_scatter_plot(cls, plot_df):
         return json.dumps(fig, cls=plotly.utils.PlotlyJSONEncoder)
 
     @classmethod
-    def create_tsne_input_space_plot(cls, features_df, candidate_or_target):
-        tsne = TSNE(n_components=2, verbose=1, perplexity=40, n_iter=300, random_state=1000)
-        tsne_result = tsne.fit_transform(features_df)
+    def create_tsne_input_space_plot(cls, plot_df):
+        # The perplexity cannot be bigger than the number of data points (the length of the dataframe).
+        # Prevent this error case by picking the smallest of the two.
```

Comment:
I am still getting the error:
ValueError: perplexity must be less than n_samples